test/libsolidity/semanticTests/variables/delete_transient_state_variable.sol

```diff
@@ -8,6 +8,5 @@ contract C {
 }
 // ====
 // EVMVersion: >=cancun
-// compileViaYul: false
 // ----
 // f() -> 0
```

Comment:
First of all, that's a really good catch. Secondly it does work (tested locally with several variables packed in multiple slots with different offsets), in a somewhat convoluted way; it gets handled by the default visitor (i.e. the non storage one):
```c++
[&](auto const&) {
	IRVariable zeroValue(m_context.newYulVariable(), m_currentLValue->type);
	define(zeroValue) << m_utils.zeroValueFunction(m_currentLValue->type) << "()\n";
	writeToLValue(*m_currentLValue, zeroValue);
	m_currentLValue.reset();
}
```
This creates a new Yul variable of the correct type and sets it to zero, and then it takes the originally passed non `IRLValue::Storage` `_lvalue` (in this case transient), and passes it off to `writeLValue(...)`, which *does* implement the update of transient storage values functionality (obviously, otherwise this PR would be kinda pointless :)).

In any case, whilst this *does* work properly, it it less efficient than extending the current behaviour for setting storage variables to zero. So, I'll extend the `YulUtilFunctions::storageSetToZeroFunction(Type const& _type)` to accept a variable location and add a transient storage visitor to the unary op visitation.